### PR TITLE
[pred_mod] refactor dataset processing

### PR DESCRIPTION
**Pull request by KaiWaldrant**
## Describe your changes
* Refactor new id in `process_dataset` script
* Refactor `process_dataset`script to add modality swapping

## Issue ticket number and link
Closes #xxxx (Replace xxxx with the GitHub issue number)

## Checklist before requesting a review
- [x] I have performed a self-review of my code

- Check the correct box. Does this PR contain:
  - [ ] Breaking changes
  - [ ] New functionality
  - [ ] Major changes
  - [x] Minor changes
  - [ ] Bug fixes

- [ ] Proposed changes are described in the CHANGELOG.md

- [ ] CI Tests succeed and look good!

## Requirements after merging

- [ ] Need to regenerate `common/` resources

- [x] Need to regenerate task-specific resources. Specify: predict_modality

### CODE REVIEW

**Comment by KaiWaldrant** [comment]
Adding this solution saves the test resources with folloing path `/openproblems_neurips2021_normal/bmmc_cite` instead of  `resources_test/predict_modality/openproblems_neurips2021/bmmc_cite_normal`

File: src/tasks/predict_modality/workflows/process_datasets/main.nf [hunk 1]
```diff
@@ -61,15 +75,17 @@ workflow run_wf {
...
+      def swap_state = state.direction == "swap" ? true : false
+      [
+        input_mod1: state.dataset_mod1,
+        input_mod2: state.dataset_mod2,
+        output_train_mod1: state.output_train_mod1,
+        output_train_mod2: state.output_train_mod2,
+        output_test_mod1: state.output_test_mod1,
+        output_test_mod2: state.output_test_mod2,
+        swap: swap_state
+      ]},
```
Comment (rcannood) [inline src/tasks/predict_modality/workflows/process_datasets/main.nf:88]: Reformat

```suggestion
      fromState: { id, state ->
        def swap_state = state.direction == "swap" ? true : false
        [
          input_mod1: state.dataset_mod1,
          input_mod2: state.dataset_mod2,
          output_train_mod1: state.output_train_mod1,
          output_train_mod2: state.output_train_mod2,
          output_test_mod1: state.output_test_mod1,
          output_test_mod2: state.output_test_mod2,
          swap: swap_state
        ]
      },
```

File: src/tasks/predict_modality/workflows/process_datasets/main.nf [hunk 2]
```diff
@@ -12,7 +12,21 @@ workflow run_wf {
...
+
   output_ch = input_ch
+    | combine(direction)
+
+    // Add swap direction to the state
+    | map{id, state, dir -> 
+      // Set new id with direction
+      def id_split = id.tokenize("/")
+      def norm = id_split.takeRight(1)[0]
+      def new_id = id_split.dropRight(1).join("/") + "_" + dir + "/" + norm
```
Comment (rcannood) [inline src/tasks/predict_modality/workflows/process_datasets/main.nf:26]: Maybe add a comment to make it clear what's happening

```suggestion
      // Add direction (normal / swap) to id
      // Note: this id is added before the normalisation id
      // Example old id: dataset_loader/dataset_id/normalization_id
      // Example new id: dataset_loader/dataset_id/direction/normalization_id
      def id_split = id.tokenize("/")
      def norm = id_split.takeRight(1)[0]
      def new_id = id_split.dropRight(1).join("/") + "_" + dir + "/" + norm
```

**Review by rcannood — APPROVED** [review]
Minor suggestion

**Pull request merged**